canonical-web-and-design/jaas-dashboard
# Persist login credentials between refresh

## Done

After a successful login, persist the macaroons between refreshes in the users localStorage.

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- You should be presented with the login link. Click to log in and it should list your models, update the statuses and display the details on the details pages.
- Refresh the page, it should instantly start loading your models without a login screen. It should again work as expected.
- In the browser console run `localStorage.clear()` 
- Refresh, you should get the login screen again, log in and test around.
- Refresh, it should no longer present you with the login screen and it should work as expected.

## Details

Now that we are persisting the users credentials we need some way to log out. Currently this can be done by opening up the developer tools in the browser and running `localStorage.clear()`. I've created a follow-up issue here: https://github.com/canonical-web-and-design/jaas-dashboard/issues/77

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/946
